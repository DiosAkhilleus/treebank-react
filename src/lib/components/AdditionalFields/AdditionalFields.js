import React from 'react';
import { arrayOf, string } from 'prop-types';

import { wordType } from '../../types';

import styles from './AdditionalFields.module.scss';

const defaultFields = [
  'sg',
  'comment',
  'artificial',
  'gloss',
  'alternateGloss',
  'semanticRole',
  'include',
  'multiword',
  'notes',
];

const fieldsFromJson = ({ $ }, fields) => (
  fields.filter((f) => Object.prototype.hasOwnProperty.call($, f)).map((f) => [f, $[f]])
);

const renderField = ([name, value]) => (
  <div key={name} className={styles.container}>
    <dt className={styles.dt}>{name}</dt>
    <dd className={styles.dd}>{value}</dd>
  </div>
);

const AdditionalFields = ({ active, fields }) => {
  if (active) {
    const displayFields = fieldsFromJson(active, fields);

    if (displayFields.length !== 0) {
      return (
        <div className={styles.information}>
          <dl className={styles.dl}>
            {displayFields.map(renderField)}
          </dl>
        </div>
      );
    }
  }

  return (
    <div />
  );
};

AdditionalFields.propTypes = {
  active: wordType,
  fields: arrayOf(string),
};

AdditionalFields.defaultProps = {
  active: null,
  fields: defaultFields,
};

export default AdditionalFields;
