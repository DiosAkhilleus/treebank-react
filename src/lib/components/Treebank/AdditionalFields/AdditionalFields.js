import React from 'react';

import SentenceContext from '../sentence-context';

import AdditionalFields from '../../AdditionalFields';

const AdditionalFieldsWithContext = () => (
  <SentenceContext.Consumer>
    {({ active }) => <AdditionalFields active={active} />}
  </SentenceContext.Consumer>
);

export default AdditionalFieldsWithContext;
