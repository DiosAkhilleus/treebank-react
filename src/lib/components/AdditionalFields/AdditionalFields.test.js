import React from 'react';
import renderer from 'react-test-renderer';

import AdditionalFields from './AdditionalFields';

it('renders a word with additional fields', () => {
  const active = {
    $: {
      id: '1',
      sg: 'sg',
      comment: 'comment',
      artificial: 'artificial',
      gloss: 'gloss',
      alternateGloss: 'alternateGloss',
      semanticRole: 'semanticRole',
      include: 'include',
      multiword: 'multiword',
      notes: 'notes',
    },
  };

  const component = (
    <AdditionalFields active={active} />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render a subset of fields', () => {
  const active = {
    $: {
      id: '1',
      sg: 'sg',
      comment: 'comment',
      artificial: 'artificial',
      gloss: 'gloss',
      alternateGloss: 'alternateGloss',
      semanticRole: 'semanticRole',
      include: 'include',
      multiword: 'multiword',
      notes: 'notes',
    },
  };

  const component = (
    <AdditionalFields active={active} fields={['comment', 'gloss']} />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders an empty div when there are no fields', () => {
  const active = { $: { id: '1' } };

  const component = (
    <AdditionalFields active={active} />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
