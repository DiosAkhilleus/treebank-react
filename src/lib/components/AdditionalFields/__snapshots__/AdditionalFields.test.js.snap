// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a subset of fields 1`] = `
<div
  className="information"
>
  <dl
    className="dl"
  >
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        comment
      </dt>
      <dd
        className="dd"
      >
        comment
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        gloss
      </dt>
      <dd
        className="dd"
      >
        gloss
      </dd>
    </div>
  </dl>
</div>
`;

exports[`renders a word with additional fields 1`] = `
<div
  className="information"
>
  <dl
    className="dl"
  >
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        sg
      </dt>
      <dd
        className="dd"
      >
        sg
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        comment
      </dt>
      <dd
        className="dd"
      >
        comment
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        artificial
      </dt>
      <dd
        className="dd"
      >
        artificial
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        gloss
      </dt>
      <dd
        className="dd"
      >
        gloss
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        alternateGloss
      </dt>
      <dd
        className="dd"
      >
        alternateGloss
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        semanticRole
      </dt>
      <dd
        className="dd"
      >
        semanticRole
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        include
      </dt>
      <dd
        className="dd"
      >
        include
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        multiword
      </dt>
      <dd
        className="dd"
      >
        multiword
      </dd>
    </div>
    <div
      className="container"
    >
      <dt
        className="dt"
      >
        notes
      </dt>
      <dd
        className="dd"
      >
        notes
      </dd>
    </div>
  </dl>
</div>
`;

exports[`renders an empty div when there are no fields 1`] = `<div />`;
